src/channel/channel.ts

```diff
@@ -70,8 +70,8 @@ export class Channel<T> implements IChannel<T> {
   ) {
     Channel.chanMap.set(this.selectSymbol, this)
 
-    typeCheck('object', config)
-    typeCheck('number?', config.bufferSize)
+    check.assert.object(config)
+    check.assert.number(config.bufferSize)
```

Comment:
I believe this is the wrong type. The bufferSize is optional for this class, since the channel can toggle between buffered and non-buffered. So the check-types equivalent should be: `check.assert.maybe.number`.